### DISCUSSION TITLE
Execute mutation based on wifi/cellular connection

### DISCUSSION THREAD

**Discussion by momme-rtf** [TanStack/query#7047] [Ideas]
We're working on a react native app in which we want to organize long running mutations with react query. The users should be able to configure whether to allow the execution only with wifi or also with a cellular internet connection. Is there a way to configure certain mutations not only based on the online/offline state but also the type of connection? If not would it make sense to extend react query with it?

**Comment by TkDodo** [comment]
you can define what counts as online / offline by setting a different event handler for the `onlineManager`

**Comment by momme-rtf** [comment]
However, this would then affect all queries and mutations at the same time. We would like to block only specific mutations/queries (while the user is using cellular) that would lead to high mobile data consumption.

**Reply by TkDodo** [reply]
I think with the current possibilities, you can only:
- overwrite the onlineManager to return false for cellular or offline, true for "good connectivity" upwards
- Then set the default networkMode to `always` and to `online` for those that need good connectivity.

the drawback is that all other queries will always run (even if offline), but the onlineManager is global and only either/or, so this is the best I could come up with
